webpack/assets/javascripts/react_app/components/hosts/PowerStatus.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import {Label} from 'react-bootstrap';
+import helpers from '../../common/helpers';
+import HostsStore from '../../stores/HostsStore';
+import HostsActions from '../../actions/HostsActions';
+
+const PowerStates = {
+    on: {
+        text: ('On'),
+        bsStyle: 'success'
+    },
+    off: {
+        text: ('Off'),
+        bsStyle: 'default'
+    },
+    na: {
+        text: ('N/A'),
+        bsStyle: 'info'
+    }
+};
+
+class PowerStatus extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = PowerStates.na; // Sets initiale state to unknown - revisit
+    helpers.bindMethods(this, [
+      'onChange',
+      'onError']
+    );
+  }
+
+  componentDidMount() {
+    HostsStore.addChangeListener(this.onChange);
+    HostsStore.addErrorListener(this.onError);
+    HostsActions.getHostPowerState(this.props.id);
+  }
+
+  componentWillUnmount() {
+    HostsStore.removeChangeListener(this.onChange);
+    HostsStore.removeErrorListener(this.onError);
+  }
+
+  onChange(event) {
+    const id = parseInt(event.id, 10);
+
+    if (id === this.props.id) {
+      const host = HostsStore.getHostsData(id);
+
+      this.updateState(host);
+    }
+  }
+
+  onError(info) {
+    console.log(info);
+    // const xhr = info.jqXHR;
+    // const id = xhr.originalRequestOptions.url.split('/')[1];
+  }
+
+  updateState(host) {
+    this.setState(PowerStates[host.power.state]);
+  }
+
+  render() {
+    return (
+      <Label bsStyle = {this.state.bsStyle}>
+        {__(this.state.text)}
```

Comment:
btw: I don't think this will extract the string correctly, but adding it to the const above fails as Jed is not initialized yet at that stage.